components/encryption/src/manager/mod.rs

```diff
@@ -477,6 +477,10 @@ impl DataKeyManager {
             // Encryption is being enabled.
             (Ok(None), _) => {
                 info!("encryption is being enabled. method = {:?}", args.method);
+                // If `master_key` doesn't work correctly, dict files will not be created.
+                let plaintext_temp = "";
+                let _ = master_key.encrypt(plaintext_temp.as_bytes())?;
```

Comment:
I think this is not a good fix. It doesn't prevent us from intermittent failure. For example the test call here could success, while the real call fail. In that case we still leave behind an empty key dict.

Reply:
Empty file dict is considered as non-existent a `file.dict` now.